bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/model/ReasonType.kt

```diff
@@ -1,16 +1,23 @@
 package io.bucketeer.sdk.android.internal.model
 
-enum class ReasonType(val value: Int) {
-  TARGET(0),
-  RULE(1),
-  DEFAULT(3),
-  CLIENT(4),
-  OFF_VARIATION(5),
-  PREREQUISITE(6),
+enum class ReasonType {
+  TARGET,
+  RULE,
+  DEFAULT,
+  CLIENT,
+  OFF_VARIATION,
+  PREREQUISITE,
 
   ;
 
   companion object {
-    fun from(value: Int): ReasonType = values().first { it.value == value }
+    fun from(value: String): ReasonType {
+      return try {
+        values().first { it.name == value }
+      } catch (ex: NoSuchElementException) {
+        // FALLBACK
+        DEFAULT
+      }
+    }
```

Comment:
You can use `firstOrNull` instead

```suggestion
    fun from(value: String): ReasonType = values().firstOrNull { it.name == value } ?? DEFAULT
```

Reply:
its much short and better